packages/framework-provider-azure-infrastructure/src/infrastructure/synth/web-pubsub-extension-key/terraform-sleep.ts

```diff
@@ -0,0 +1,22 @@
+import { TerraformStack } from 'cdktf'
+import { functionAppFunction } from '@cdktf/provider-azurerm'
+import { toTerraformName } from '../../helper/utils'
+import { sleep } from '@cdktf/provider-time'
+import { TimeProvider } from '@cdktf/provider-time/lib/provider'
+
+export class TerraformSleep {
+  static build(
+    terraformStackResource: TerraformStack,
+    appPrefix: string,
+    dummyFunction: functionAppFunction.FunctionAppFunction
+  ): sleep.Sleep {
+    const timeProvider = new TimeProvider(terraformStackResource, 'sleepFeature', {})
+
+    const id = toTerraformName(appPrefix, 'sl')
+    return new sleep.Sleep(terraformStackResource, id, {
+      createDuration: '5m',
+      dependsOn: [dummyFunction],
```

Comment:
A bit nitpicking:
I guess this "sleep" feature is used to be able to get the key used in the hub. I know we use a dummy function for this but, in the context of this build function, it doesn't make sense that this parameter is called "dummyFunction", as we can use this sleep functionality for any function.
Mainly I mean to remove the `dummy` prefix.

Reply:
What is more: Can we sleep depending on any other resource, not just functions? If so, we could change the time of the last parameter to be a generic resource type, allowing us to reuse this function whenever we need to wait for any resource